src/pages/DataLoader/DataLoader.tsx

```diff
@@ -0,0 +1,251 @@
+        <input
+          accept=".json"
+          type="file"
+          onChange={(e) => {
+            setFile(e.target.files[0]);
+            setFileName(e.target.files[0].name);
+          }}
+        />
+      </div>
+    );
+  };
+
+  const renderFileNameField = () => {
+    return (
+      <div className="data-loader__input">
+        <label htmlFor="data-loader-file-name">File name</label>
+        <input
+          id="data-loader-file-name"
+          type="text"
+          value={fileName}
+          onChange={(e) => setFileName(e.target.value)}
+        />
+      </div>
+    );
+  };
+
+  const validateFileName = () => {
+    let valid = true;
+    // access-policy.json would cause the backend to break
+    if (fileName.includes('access-policy')) {
+      setMessage('File name cannot include access-policy');
+      valid = false;
+    }
+    if (!fileName) {
+      setMessage('File name cannot be empty');
+      valid = false;
+    }
+    return valid;
+  };
+
+  const renderButton = () => {
+    const buttonData = {
+      validate: {
+        method: validate,
+        text: 'Validate',
+      },
+      upload: {
+        method: upload,
+        text: 'Upload',
+      },
+      harvest: {
+        method: harvest,
+        text: 'Harvest',
+      },
+    };
+
+    return (
+      <Button disabled={running} onClick={buttonData[state].method}>
+        {running ? 'Running' : buttonData[state].text}
+      </Button>
+    );
+  };
+
+  const validate = async () => {
+    if (!file) {
+      setMessage('Please select a file before running validation');
+      return;
+    }
+
+    setRunning(true);
+    setMessage('Validating STAC file');
+    try {
+      await validateSTAC();
+      setMessage('File successfully validated');
+      setState('upload');
+    } catch (error) {
+      console.error(error);
+    }
+    setRunning(false);
+  };
+
+  const validateSTAC = async () => {
+    const stacContent = await file.text();
+    let stac;
+    try {
+      stac = JSON.parse(stacContent);
+    } catch (e) {
+      setMessage('❌ Invalid JSON format');
+      throw new Error();
+    }
+
+    let ajv;
+
+    try {
+      ajv = new Ajv({ strict: false });
+      addFormats(ajv);
+      ajv.addMetaSchema(draft07);
+      ajv.addSchema(feature, 'https://geojson.org/schema/Feature.json');
+      ajv.addSchema(geometry, 'https://geojson.org/schema/Geometry.json');
+      ajv.addSchema(
+        basics,
+        'https://schemas.stacspec.org/v1.0.0/item-spec/json-schema/basics.json',
+      );
+      ajv.addSchema(
+        datetime,
+        'https://schemas.stacspec.org/v1.0.0/item-spec/json-schema/datetime.json',
+      );
+      ajv.addSchema(
+        instrument,
+        'https://schemas.stacspec.org/v1.0.0/item-spec/json-schema/instrument.json',
+      );
+      ajv.addSchema(
+        licensing,
+        'https://schemas.stacspec.org/v1.0.0/item-spec/json-schema/licensing.json',
+      );
+      ajv.addSchema(
```

Comment:
Do all of these schema's have to validate? I know that a lot of these are optional, as many come from STAC Extensions. 

https://stac-extensions.github.io/